### PR TITLE
[SM-191] Create ServiceAccount Table

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The purpose of this PR is to create the Service Account database table for the Secret Manager project.

## Code changes

- bitwarden_license/src/Commercial.Infrastructure.EntityFramework/CommercialEFServiceCollectionExtensions.cs

Add ServiceAccountRepository into dependency injection

- bitwarden_license/src/Commercial.Infrastructure.EntityFramework/Repositories/ServiceAccountRepository.cs
src/Core/Repositories/IServiceAccountRepository.cs

ServiceAccountRepository implementation and interface.
GetById, GetManyByOrganization, Create, Update

- src/Core/Entities/ServiceAccount.cs

Service Account entity

- src/Infrastructure.EntityFramework/Configurations/ServiceAccountEntityTypeConfiguration.cs
src/Infrastructure.EntityFramework/Models/ServiceAccount.cs

Entity Framework Service Account entity mapping and configuration.

- src/Infrastructure.EntityFramework/Repositories/DatabaseContext.cs

Entity Framework add Service Account table into database context.

- src/Sql/dbo/Tables/ServiceAccount.sql
util/Migrator/DbScripts/2022-09-26_00_SmServiceAccount.sql

Microsoft SQL Server migrations

- util/MySqlMigrations/Migrations/20220926191322_SmServiceAccount.Designer.cs util/MySqlMigrations/Migrations/20220926191322_SmServiceAccount.cs util/MySqlMigrations/Migrations/DatabaseContextModelSnapshot.cs util/MySqlMigrations/Scripts/2022-09-26_00_SmServiceAccount.sql 

MySQL migrations

- util/PostgresMigrations/Migrations/20220926190851_SmServiceAccount.Designer.cs util/PostgresMigrations/Migrations/20220926190851_SmServiceAccount.cs util/PostgresMigrations/Migrations/DatabaseContextModelSnapshot.cs util/PostgresMigrations/Scripts/2022-09-26_00_SmServiceAccount.psql

Postgres migrations

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: bitwarden_license/src/Commercial.Infrastructure.EntityFramework/Repositories/ServiceAccountRepository.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Bit.Commercial.Infrastructure.EntityFramework.Repositories
+{
+    public class ServiceAccountRepository : Repository<Core.Entities.ServiceAccount, ServiceAccount, Guid>, IServiceAccountRepository
+    {
+        public ServiceAccountRepository(IServiceScopeFactory serviceScopeFactory, IMapper mapper)
+            : base(serviceScopeFactory, mapper, db => db.ServiceAccount)
+        {
+
+        }
```
Comment (Hinton) [inline bitwarden_license/src/Commercial.Infrastructure.EntityFramework/Repositories/ServiceAccountRepository.cs:16]:
```suggestion
        { }
```

File: util/PostgresMigrations/Scripts/2022-09-26_00_SmServiceAccount.psql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    CONSTRAINT "PK_ServiceAccount" PRIMARY KEY ("Id"),
+    CONSTRAINT "FK_ServiceAccount_Organization_OrganizationId" FOREIGN KEY ("OrganizationId") REFERENCES "Organization" ("Id") ON DELETE CASCADE
+);
+
+CREATE INDEX "IX_ServiceAccount_OrganizationId" ON "ServiceAccount" ("OrganizationId");
+
+INSERT INTO "__EFMigrationsHistory" ("MigrationId", "ProductVersion")
+VALUES ('20220926190851_SmServiceAccount', '6.0.4');
+
+COMMIT;
```
Comment (Hinton) [inline util/PostgresMigrations/Scripts/2022-09-26_00_SmServiceAccount.psql:18]:
```suggestion
COMMIT;

```

**Review by Hinton — COMMENTED** [review]
Just a few minor syntax changes

File: bitwarden_license/src/Commercial.Infrastructure.EntityFramework/Repositories/ServiceAccountRepository.cs (hunk 1 above)
Comment (Thomas-Avery) [reply bitwarden_license/src/Commercial.Infrastructure.EntityFramework/Repositories/ServiceAccountRepository.cs]: Went ahead and made this formatting change in Secret and Project repos as well to keep consistent. 

Resolved in https://github.com/bitwarden/server/pull/2301/commits/e2746d040e644a4550ba4d15595dabc375a8adf8

File: util/PostgresMigrations/Scripts/2022-09-26_00_SmServiceAccount.psql (hunk 2 above)
Comment (Thomas-Avery) [reply util/PostgresMigrations/Scripts/2022-09-26_00_SmServiceAccount.psql]: Resolved in https://github.com/bitwarden/server/pull/2301/commits/e2746d040e644a4550ba4d15595dabc375a8adf8

**Review by Hinton — APPROVED** [review]

**Pull request merged**